.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,31 @@
+# .github/workflows/shellcheck.yml
+---
+name: shellcheck
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: ["main", "master", "stable", "feature-exploring-sarif-output-165"]
+  pull_request:
+    branches: ["main", "master", "stable", "feature-*", "patch-*", "HOTFIX-*"]
+
+permissions: {}  # Setting default permissions to none for enhanced security
+
+jobs:
+  shellcheck:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
+      actions: read
+      pull-requests: read
+      statuses: write
+      security-events: write  # for github/codeql-action/upload-sarif to upload SARIF results
```

Comment:
@coderabbitai, :confused: Is the `actions: read` not inherited by the composite action `shellcheck-scan` in this specific case? It is not visible in the workflow but the read actions permission is use in shellcheck-scan if I'm not mistaken. 🤷 

The statuses permission is to ensure the outcome of the workflow is written to the pushed commits afterwards, do you mean to say that it will still pass/fail the commit statutes on pushing without the statuses permission?! Please elaborate.

I don't want extra permissions, but I don't want to be guessing either.